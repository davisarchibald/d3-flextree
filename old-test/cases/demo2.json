{
  "name": "root",
  "x_size": 15,
  "y_size": 80,
  "children": [
    {
      "name": "long",
      "x_size": 15,
      "y_size": 150,
      "children": [
        {
          "name": "long",
          "x_size": 15,
          "y_size": 150,
          "children": [
            { 
              "name": "leaf",
              "x_size": 200,
              "y_size": 80
            },
            { 
              "name": "short",
              "x_size": 15,
              "y_size": 80,
              "children": [
                {
                  "name": "leaf",
                  "x_size": 15,
                  "y_size": 80
                }
              ]
            }
          ]
        }
      ]
    },
    {
      "name": "short",
      "x_size": 15,
      "y_size": 80,
      "children": [
        {
          "name": "short",
          "x_size": 15,
          "y_size": 80,
          "children": [
            { 
              "name": "short",
              "x_size": 15,
              "y_size": 80,
              "children": [
                {
                  "name": "long",
                  "x_size": 15,
                  "y_size": 300,
                  "children": [
                    { 
                      "name": "leaf",
                      "x_size": 15,
                      "y_size": 80
                    },
                    { 
                      "name": "leaf",
                      "x_size": 15,
                      "y_size": 80
                    },
                    { 
                      "name": "leaf",
                      "x_size": 15,
                      "y_size": 80
                    },
                    { 
                      "name": "leaf",
                      "x_size": 15,
                      "y_size": 80
                    }
                  ]
                }
              ]
            },
            { 
              "name": "leaf",
              "x_size": 15,
              "y_size": 80
            },
            { 
              "name": "leaf",
              "x_size": 15,
              "y_size": 80
            },
            { 
              "name": "leaf",
              "x_size": 15,
              "y_size": 80
            }
          ]
        }
      ]
    },
    {
      "name": "long",
      "x_size": 15,
      "y_size": 150,
      "children": [
        {
          "name": "long",
          "x_size": 15,
          "y_size": 150,
          "children": [
            { 
              "name": "leaf",
              "x_size": 15,
              "y_size": 80
            },
            { 
              "name": "leaf",
              "x_size": 300,
              "y_size": 80
            },
            { 
              "name": "leaf",
              "x_size": 15,
              "y_size": 80
            },
            { 
              "name": "leaf",
              "x_size": 15,
              "y_size": 80
            }
          ]
        }
      ]
    }
  ]
}
