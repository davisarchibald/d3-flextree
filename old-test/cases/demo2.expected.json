{
  "x" : 0.0,
  "y" : 0.0,
  "x_size" : 15.0,
  "y_size" : 80.0,
  "children" : [ {
    "x" : -162.5,
    "y" : 80.0,
    "x_size" : 15.0,
    "y_size" : 150.0,
    "children" : [ {
      "x" : -162.5,
      "y" : 230.0,
      "x_size" : 15.0,
      "y_size" : 150.0,
      "children" : [ {
        "x" : -170.0,
        "y" : 380.0,
        "x_size" : 200.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      }, {
        "x" : -62.5,
        "y" : 380.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ {
          "x" : -62.5,
          "y" : 460.0,
          "x_size" : 15.0,
          "y_size" : 80.0,
          "children" : [ ],
          "name" : "leaf"
        } ],
        "name" : "short"
      } ],
      "name" : "long"
    } ],
    "name" : "long"
  }, {
    "x" : -10.0,
    "y" : 80.0,
    "x_size" : 15.0,
    "y_size" : 80.0,
    "children" : [ {
      "x" : -10.0,
      "y" : 160.0,
      "x_size" : 15.0,
      "y_size" : 80.0,
      "children" : [ {
        "x" : -32.5,
        "y" : 240.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ {
          "x" : -32.5,
          "y" : 320.0,
          "x_size" : 15.0,
          "y_size" : 300.0,
          "children" : [ {
            "x" : -55.0,
            "y" : 620.0,
            "x_size" : 15.0,
            "y_size" : 80.0,
            "children" : [ ],
            "name" : "leaf"
          }, {
            "x" : -40.0,
            "y" : 620.0,
            "x_size" : 15.0,
            "y_size" : 80.0,
            "children" : [ ],
            "name" : "leaf"
          }, {
            "x" : -25.0,
            "y" : 620.0,
            "x_size" : 15.0,
            "y_size" : 80.0,
            "children" : [ ],
            "name" : "leaf"
          }, {
            "x" : -10.0,
            "y" : 620.0,
            "x_size" : 15.0,
            "y_size" : 80.0,
            "children" : [ ],
            "name" : "leaf"
          } ],
          "name" : "long"
        } ],
        "name" : "short"
      }, {
        "x" : -17.5,
        "y" : 240.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      }, {
        "x" : -2.5,
        "y" : 240.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      }, {
        "x" : 12.5,
        "y" : 240.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      } ],
      "name" : "short"
    } ],
    "name" : "short"
  }, {
    "x" : 162.5,
    "y" : 80.0,
    "x_size" : 15.0,
    "y_size" : 150.0,
    "children" : [ {
      "x" : 162.5,
      "y" : 230.0,
      "x_size" : 15.0,
      "y_size" : 150.0,
      "children" : [ {
        "x" : -2.5,
        "y" : 380.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      }, {
        "x" : 155.0,
        "y" : 380.0,
        "x_size" : 300.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      }, {
        "x" : 312.5,
        "y" : 380.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      }, {
        "x" : 327.5,
        "y" : 380.0,
        "x_size" : 15.0,
        "y_size" : 80.0,
        "children" : [ ],
        "name" : "leaf"
      } ],
      "name" : "long"
    } ],
    "name" : "long"
  } ],
  "name" : "root"
}
